python/tests/bins_test.py

```diff
@@ -200,3 +201,69 @@ def test_bins_sum_with_masked_buffer():
     xbins = sc.Variable(dims=['x'], unit=sc.units.m, values=[0.1, 0.5, 0.9])
     binned = sc.bin(data, edges=[xbins])
     assert binned.bins.sum().values[0] == 2
+
+
+def test_bins_mean():
+    data = sc.DataArray(data=sc.Variable(dims=['position'],
+                                         unit=sc.units.counts,
+                                         values=[0.1, 0.2, 0.3, 0.4, 0.5]),
+                        coords={
+                            'x':
+                            sc.Variable(dims=['position'],
+                                        unit=sc.units.m,
+                                        values=[1, 2, 3, 4, 5])
+                        })
+    xbins = sc.Variable(dims=['x'], unit=sc.units.m, values=[0, 5, 6, 7])
+    binned = sc.bin(data, edges=[xbins])
+
+    # Mean of [0.1, 0.2, 0.3, 0.4]
+    assert binned.bins.mean().values[0] == 0.25
+
+    # Mean of [0.5]
+    assert binned.bins.mean().values[1] == 0.5
+
+    # Mean of last (empty) bin should be NaN
+    assert isnan(binned.bins.mean().values[2])
```

Comment:
We should assert on properties such as `values` only in exceptional cases. By default, construct a reference variable and compare to that using `sc.identical`. Otherwise you easily miss important aspects in the test, such as dimensions, units, and variances.

Reply:
The issue here is that the array contains a `NaN` value, and `NaN` doesn't compare equal to itself, so can't use `sc.identical` easily as it will always return `False` if the variable contains a `NaN`.

Minimal test case:
```
>>> sc.identical(sc.scalar(float("NaN")), sc.scalar(float("NaN")))
False
```

Reply:
Ok, then you will need to add assertions for at least `dims`, `shape`, and `unit` by hand.